pp_sys.c

```diff
@@ -2989,7 +2989,13 @@ PP(pp_stat)
             Stat_t s;
             CLANG_DIAG_IGNORE_STMT(-Wtautological-compare);
             GCC_DIAG_IGNORE_STMT(-Wtype-limits);
+#if defined(__HP_cc) || defined(__HP_aCC)
+#pragma diag_suppress 2186
+#endif
             neg = PL_statcache.st_ino < 0;
+#if defined(__HP_cc) || defined(__HP_aCC)
+#pragma diag_default 2186
+#endif
             GCC_DIAG_RESTORE_STMT;
             CLANG_DIAG_RESTORE_STMT;
```

Comment:
We can avoid all of this - config.h defines `ST_INO_SIGN` which has a negative value if `st_ino` is signed.

Especially since "fsblkcnt_t, fsfilcnt_t, and ino_t shall be defined as unsigned integer types" per POSIX.

Reply:
Ok. Ill change it to use that instead. Are you thinking i should remove the DIAG_IGNORE statements as well?  I have to admit i tend to think that it would be better to merge this with the HPUX diag ignores, then afterward do a follow up that removes all of them. What do you think?

Reply:
Ok